### ISSUE TITLE
Text should be resizable

### ISSUE THREAD

**Issue by JonasJurczok** [JonasJurczok/factorio-todo-list#41]
It should be possible to change the font size.

**Comment by hoylemd** [comment]
I think it would have something to do with the [`style`](https://lua-api.factorio.com/latest/LuaGuiElement.html#LuaGuiElement.style) parameter for the `add` method used to create ui elements.  Takes a string or a [LuaStyle](https://lua-api.factorio.com/latest/LuaStyle.html) object.

I've never worked in lua before, much less modding factorio, but I've been wanting to get into it. Maybe I'll try to tackle this as a first project.

**Comment by JonasJurczok** [comment]
Sure :) feel free to open a pull request :)

Am 26.07.2018 6:53 nachm. schrieb "Michael D. Hoyle" <
notifications@github.com>:

I think it would have something to do with the style
<https://lua-api.factorio.com/latest/LuaGuiElement.html#LuaGuiElement.style>
parameter for the add method used to create ui elements. Takes a string or
a LuaStyle <https://lua-api.factorio.com/latest/LuaStyle.html> object.

I've never worked in lua before, much less modding factorio, but I've been
wanting to get into it. Maybe I'll try to tackle this as a first project.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/JonasJurczok/factorio-todo-list/issues/41#issuecomment-408163270>,
or mute the thread

**Comment by hoylemd** [comment]
in #44, I just tweaked the styles a bit, but I'm thinking this could be implemented with a setting: Choose a UI style: normal (what it is now, or what is is in #44) or compact (smaller font size/padding etc)

**Comment by JonasJurczok** [comment]
Initially I thought about a text box to choose a font size. But maybe
that's too much/not possible?!

Michael D. Hoyle <notifications@github.com> schrieb am Di., 7. Aug. 2018,
20:15:

**Comment by hoylemd** [comment]
That would work, but there's a couple different text sizes in use (So headings look like headings), and things like button height may need to adjust with it.  So all of that would have to be computed from the user-selected font size, which really expands the potential error space.  We could allow each value to be set individually, but that's a lot to ask of the average user.

But if we just had say, 3 UI sizes that we've chosen and tested, They just have to pick one and they're good to go.

**Comment by JonasJurczok** [comment]
That's a good point. I thought about leaving all the headings as they are
and just make the normal text size adjustable. If I understood the button
documentation correctly it should automatically change size with the font.

>

**Comment by Tarrke** [comment]
The work to do here is just to get a way to choose which style to use, right ?

**Comment by hoylemd** [comment]
Yeah, you could have a setting to choose a size, and then have that setting determine which style to use when creating UI elements.

**Comment by JonasJurczok** [comment]
Yep. This was requested by a user. Maybe for really high/low resolutions the current default might be a problem.

**Comment by Tarrke** [comment]
Then @JonasJurczok can you add a pririty tag so we'll fix this quickly.

**Comment by JonasJurczok** [comment]
I don't know if this is actually urgent :) I view it more as a "nice to have"

**Comment by Alexander-Serov** [comment]
**Suggested implementation plan**, following the preset consensus from the discussion above:

- Per-player dropdown setting `todolist-ui-scale` = compact / normal / large.
- `src/todo/style.lua`: derive the three style sets from one table of base sizes (font, button height, checkbox width) so they can't drift apart; name them `todo_<element>_<scale>`.
- GUI builders pick the style suffix from the player setting at render time; `on_runtime_mod_setting_changed` already rebuilds on setting change (same branch should also start honouring `todolist-window-height`, which is currently ignored on change).
- Pairs naturally with the subtask width fix in #145/#192 — capped widths must scale with the preset.
